ambry-network/src/main/java/com/github/ambry/network/http2/AmbryNetworkRequestHandler.java

```diff
@@ -32,18 +35,42 @@ public class AmbryNetworkRequestHandler extends SimpleChannelInboundHandler<Full
   private static final Logger logger = LoggerFactory.getLogger(AmbryNetworkRequestHandler.class);
   private final RequestResponseChannel requestResponseChannel;
   private final Http2ServerMetrics http2ServerMetrics;
+  private final RequestAPI requestAPI;
 
   public AmbryNetworkRequestHandler(RequestResponseChannel requestResponseChannel,
-      Http2ServerMetrics http2ServerMetrics) {
+      Http2ServerMetrics http2ServerMetrics, RequestAPI requestAPI) {
     this.requestResponseChannel = requestResponseChannel;
     this.http2ServerMetrics = http2ServerMetrics;
+    this.requestAPI = requestAPI;
   }
 
   @Override
   protected void channelRead0(ChannelHandlerContext ctx, FullHttpRequest msg) {
     ByteBuf dup = msg.content().retainedDuplicate();
     try {
-      requestResponseChannel.sendRequest(new NettyServerRequest(ctx, dup));
+      NettyServerRequest networkRequest = new NettyServerRequest(ctx, dup);
+      boolean isRequestQueued = requestResponseChannel.sendRequest(networkRequest);
+      // Reject the request if input queue is full.
+      if (!isRequestQueued) {
+        try {
+          // 1. Consume the first 8 bytes of the input stream which represents the size of the request. This is added to the
+          // request since Ambry's SocketServer stack needs to know the number of the bytes to read for a request
+          // (see BoundedNettyByteBufReceive#readFrom(ReadableByteChannel) method). This is not needed when using
+          // Netty HTTP2 server stack. We can remove this once SocketServer stack is retired. For now, we are
+          // consuming the bytes here so that rest of request handling in server is same.
+          DataInputStream stream = new DataInputStream(networkRequest.getInputStream());
+          stream.readLong();
```

Comment:
Didn't understand the logic of consuming first 8 bytes. We are not using it anywhere here.